packages/next/src/utils.ts

```diff
@@ -1436,16 +1438,36 @@ export async function getPageLambdaGroups({
 
     let opts: { memory?: number; maxDuration?: number } = {};
 
+    if (functionsConfigManifest && functionsConfigManifest[routeName]) {
+      opts = functionsConfigManifest[routeName];
+    }
+
     if (config && config.functions) {
       const sourceFile = await getSourceFilePathFromPage({
         workPath: entryPath,
         page,
         pageExtensions,
       });
-      opts = await getLambdaOptionsFromFunction({
+
+      const configOpts = await getLambdaOptionsFromFunction({
         sourceFile,
         config,
       });
+
+      // If a user tries to customize in both handler and vercel.json we throw
+      const duplicateKeys = Object.keys(opts).filter(key =>
+        Object.keys(configOpts).includes(key)
```

Comment:
You should move `Object.keys(configOpts)` out of the loop so it only runs once.

Also if there are a lot of keys, might be best to use `new Set(Object.keys(configOpts))`